src/framework/mpas_pool_routines.F

```diff
@@ -5992,5 +5992,30 @@ integer function pool_get_member_decomp_type(dimName) result(decompType)!{{{
 
    end function pool_get_member_decomp_type!}}}
 
+   subroutine pool_hash(hash, key)!{{{
+
+      use iso_c_binding, only : c_int, c_char
+      use mpas_c_interfacing, only : mpas_f_to_c_string
+
+      implicit none
+
+      interface
+        subroutine c_pool_hash(hash, key) bind(c)
+           use iso_c_binding, only : c_int, c_char
+           integer (c_int), intent(inout) :: hash
+           character (c_char), intent(in) :: key
+        end subroutine c_pool_hash
+      end interface
+
+      integer (c_int), intent(inout) :: hash
+      character(len=*), intent(in) :: key
+
+      character(len=StrKIND, kind=c_char) :: c_key
+
+      call mpas_f_to_c_string(key, c_key)
```

Comment:
Since we're now copying the Fortran string to a C character array, it might be interesting to check the timing to see whether this impacts the performance of the pool hashing in any significant way. It could that, if Fortran's `len_trim` is an O(n) operation, there's essentially no impact in the O(n) operation of copying to a null-terminated C array.

Reply:
As we said early, and for the PR. The original `pool_hash` function ran slightly faster then `mpas_f_to_c_string`, but `mpas_f_to_c_string` was faster then the intrinsics `len` and `len(adjustl)`.